CMP-3859: Add e2e test (60342) for disabled timeout with zero value

taimurhafeez opened the pull request:
test case to verify that setting the timeout to 0s in a ScanSetting disables the timeout functionality. This ensures scans can complete without timeout errors when timeout is explicitly set to zero.

The test creates a ScanSetting with a timeout set to 0 seconds and a maxRetryOnTimeout set to 2, binds it to the ocp4-cis profile, and verifies that all scans complete successfully in the DONE phase without any ERROR results due to timeouts.

To implement this PR:

 `make e2e-parallel E2E_GO_TEST_FLAGS="-v -run TestTimeoutDisabledWithZeroValue"`


Tested on OCP 4.22 and expected output:
```
=== RUN   TestTimeoutDisabledWithZeroValue
=== PAUSE TestTimeoutDisabledWithZeroValue
=== CONT  TestTimeoutDisabledWithZeroValue
2026/04/24 10:56:31 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:35 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:40 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:45 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:50 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:56:55 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: RUNNING
2026/04/24 10:57:01 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: AGGREGATING
2026/04/24 10:57:06 waiting until suite test-timeout-disabled-with-zero-value-binding reaches target status 'DONE'. Current status: AGGREGATING
2026/04/24 10:57:16 ComplianceScan ready (DONE)
2026/04/24 10:57:16 All scans in ComplianceSuite have finished (test-timeout-disabled-with-zero-value-binding)
--- PASS: TestTimeoutDisabledWithZeroValue (51.85s)
PASS

```

taimurhafeez committed:
Add e2e test for disabled timeout with zero value

Port test case OCP-60342 to verify that setting timeout to 0s
in a ScanSetting disables the timeout functionality. This ensures
scans can complete without timeout errors when timeout is explicitly
set to zero.

The test creates a ScanSetting with timeout set to 0s and
maxRetryOnTimeout set to 2, binds it to the ocp4-cis profile,
and verifies that all scans complete successfully in DONE phase
without ERROR results from timeouts.

This addresses the missing upstream test noted in the downstream
test suite.

taimurhafeez committed:
Two changes: 1. Add assertion to verify scans do not have the timeout annotation
   (compliance.openshift.io/timeout) when timeout is disabled with 0s

2. Move TestTimeoutDisabledWithZeroValue from serial to parallel folder

Related: OCP-60342

xiaojiey commented on tests/e2e/framework/main_entry.go:
```diff
@@ -123,6 +123,7 @@ func (f *Framework) SetUp() error {
 	err = f.createInvalidMachineConfigPool("e2e-invalid")
 	if err != nil {
 		return fmt.Errorf("failed to create Machine Config Pool %s: %w", "e2e-invalid", err)
+		//log.Printf("did not find machineconfig e2e-invalid, but still proceeding")
```
Delete comment

taimurhafeez committed:
Removed whitespaces and downstream test ID

taimurhafeez commented on tests/e2e/framework/common.go:
```diff
@@ -818,6 +818,7 @@ func (f *Framework) createInvalidMachineConfigPool(n string) error {
 		})
 	if createErr != nil {
 		return fmt.Errorf("failed to create Machine Config Pool %s: %w", n, createErr)
+		//log.Printf("did not find machineconfig e2e-invalid, but still proceeding")
```
done.

taimurhafeez commented on tests/e2e/framework/common.go:
```diff
@@ -831,11 +832,13 @@ func (f *Framework) cleanUpMachineConfigPool(n string) error {
 	err := f.Client.Get(context.TODO(), types.NamespacedName{Name: n}, p)
 	if err != nil {
 		return fmt.Errorf("failed to get Machine Config Pool %s for cleanup: %w", n, err)
+
```
removed.

rhmdnd reviewed (commented):
Looks good to me, just need to remove the comment in main_entry.go

taimurhafeez committed:
removed commented log line

rhmdnd reviewed (approved):
/lgtm

rhmdnd commented:
/retest-required

Looks like some of the scans timed out... We might want to consider breaking the test suite into smaller pieces based on functionality. That might help us isolate problematic tests.

rhmdnd commented:
> > Another issue with a scan timing out. This time, waiting for the compliance suite to be created, which seems odd..
> > ```
> > 2025/12/03 00:41:05 waiting for availability of test-suspend-scan-setting-binding compliancesuite
> > 2025/12/03 00:41:05 waiting for availability of test-suspend-scan-setting-binding compliancesuite
> >     main_test.go:1770: compliancesuites.compliance.openshift.io "test-suspend-scan-setting-binding" not found
> > --- FAIL: TestSuspendScanSetting (1800.44s) 
> > ```
> 
> Strange. Is it from the CI run or your local run?

From CI - https://storage.googleapis.com/test-platform-results/pr-logs/pull/ComplianceAsCode_compliance-operator/982/pull-ci-ComplianceAsCode-compliance-operator-master-e2e-aws-serial/1995970267255410688/build-log.txt

rhmdnd reviewed (approved):
/lgtm

rhmdnd reviewed (approved):
Only minor comments about cosmetic issues that we can cleanup in a follow on PR.

rhmdnd commented:
I wonder if this needs to be rebased to pass enterprise contract testing?

taimurhafeez committed:
Clean up TestTimeoutDisabledWithZeroValue test

Remove MaxRetryOnTimeout setting and redundant status checks that are
already performed by WaitForSuiteScansStatus.

rhmdnd reviewed (commented):
Looks like we may have a potential issue with the test:

```
2026/03/03 15:21:55 Waiting for availability of ocp4-cis compliancescan
2026/03/03 15:21:55 Waiting for availability of ocp4-cis compliancescan
2026/03/03 15:21:55 All scans in ComplianceSuite have finished (test-timeout-disabled-with-zero-value-binding)
=== NAME  TestTimeoutDisabledWithZeroValue
    main_test.go:5420: failed to get scan ocp4-cis: compliancescans.compliance.openshift.io "ocp4-cis" not found
--- FAIL: TestTimeoutDisabledWithZeroValue (1815.07s)
FAIL
FAIL	github.com/ComplianceAsCode/compliance-operator/tests/e2e/parallel	2162.543s
```

rhmdnd commented:
Putting this back on my review queue until we can investigate the potential issue with the new test.

taimurhafeez committed:
rebased with master

rhmdnd reviewed (approved):
Looks good to me. The only thing to note is the whitespace changes across `crd.yaml`, `deploy_rbac.yaml`, and `CheckType` in `main_test.go`. I like to keep whitespace separate from functional changes since it keeps the git history cleaner, and makes it easier to parse the history of a file.
